### ISSUE TITLE
Overzealous warning when locations are configured per environment

### ISSUE THREAD

**Issue by tdg5** [flyway/flyway#3965]
###### Which version and edition of Flyway are you using?
10.19.0

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
CLI

###### Which database are you using? (Type & version)
SQL Server, but I'd be shocked if this issue had anything to do with DB vendor.

###### Which operating system are you using?
MacOS 14.6.1 (23G93), but running `flyway` docker container `flyway/flyway:10.19.0`

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)

I have chosen to configure `locations` per environment and not in the root `flyway` config.

e.g. `flyway.toml`
```toml
[flyway]
target = 'latest'

[environments.whatever]
flyway.locations = ["filesystem:/flyway/migrations/whatever"]
```

I can suppress the warning by setting a dummy value for `flyway.locations`:

```toml
[flyway]
# Override default locations to silence inaccurate warning. Actual locations
# are set per environment.
locations = ['ignored-and-overridden-per-environment']
target = 'latest'

[environments.whatever]
flyway.locations = ["filesystem:/flyway/migrations/whatever"]
```

###### What did you expect to see?
No warning of `WARNING: No locations configured and default location 'sql' not found.`

###### What did you see instead?
Warning of `WARNING: No locations configured and default location 'sql' not found.`, however migrations are detected and run correctly.

**Comment by JasonLuo-Redgate** [comment]
Hi @tdg5 ,
Many thanks for reporting this issue, we're looking into it.

**Comment by JasonLuo-Redgate** [comment]
Hi @tdg5 ,
The fix has been included the latest release 10.20.1 (https://documentation.red-gate.com/flyway/release-notes-and-older-versions/release-notes-for-flyway-engine), hence closing this ticket.
Thanks again for reporting this issue.

**Comment by will-e-yams** [comment]
I am still seeing this behavior in `Flyway Community Edition 11.18.0-rc1131 by Redgate`.

Which version and edition of Flyway are you using?
`Flyway Desktop Community Version 8.6.2`

Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
`CLI` (packed alongside desktop, I believe)

Which database are you using? (Type & version)
`Postgres`

Which operating system are you using?
`Windows 11 Pro v25H2`

```
PS C:\Users\...\some-repo\...\database> flyway -v
WARNING: No locations configured and default location 'sql' not found.
Flyway Community Edition 11.18.0-rc1131 by Redgate

See release notes here: https://rd.gt/416ObMi
Plugin Name                       | Version
--------------------------------- | ---------------
DB2 for z/OS                      | 10.24.0
OceanBase                         | 10.24.0
QuestDB                           | 10.24.0
TiDB                              | 10.24.0
YugabyteDB                        | 10.24.0
ClickHouse                        | 10.24.0
CUBRID                            | 10.24.0
Databricks                        | 10.24.0
DuckDB                            | 10.24.0
Apache Ignite                     | 10.24.0
InterSystems IRIS Data Platform   | 10.24.0
Timeplus                          | 10.24.0
Redgate Compare                   | 1.45.1.2083
SQLFluff                          | not installed
```
